google/cloud/forseti/scanner/audit/external_project_access_rules_engine.py

```diff
@@ -274,18 +304,10 @@ def find_violation(self, user_email, ancestry):
             namedtuple: Returns RuleViolation named tuple or None if
                 not violated.
         """
-        matched_resources = []
-        applies_to_user = True
-
-        for resource in self.rules['ancestor_resources']:
-            if resource in ancestry:
-                matched_resources.append(resource)
-
-        if 'users' in list(self.rules.keys()):
-            if user_email not in self.rules['users']:
-                applies_to_user = False
+        applies_to_user = self._applies_to_user(user_email)
+        matched_ancestry = self._matched_ancestry(ancestry)
 
-        if not (matched_resources and applies_to_user):
+        if not applies_to_user or (applies_to_user and not matched_ancestry):
```

Comment:
Just confirming that this logic is intended to be the same? Perhaps it would be simpler to express this as:
`if not applies_to_user or not matched_ancestry`?